packages/pages-shared/metadata-generator/parseRedirects.ts

```diff
@@ -109,6 +109,19 @@ export function parseRedirects(input: string): ParsedRedirects {
 			continue;
 		}
 
+		// We want to always block the `/* /index.html` redirect - this will cause TOO_MANY_REDIRECTS errors as
+		// the asset server will redirect it back to `/`, removing the `/index.html`. This is the case for regular
+		// redirects, as well as proxied (200) rewrites. We only want to run this on relative urls
+		if (/\/\*?$/.test(from) && /\/index(.html)?$/.test(to) && !urlHasHost(to)) {
+			invalid.push({
+				line,
+				lineNumber: i + 1,
+				message:
+					'This behaviour is default with Cloudflare Pages (when a 404.html isn\'t present) and will be ignored. Remove it from your _redirects to silence this warning.\nIf you wish to direct requests to a subfolder, for example, sending all requests to foo/index.html, you should use "/* /foo/".',
```

Comment:
This message is really only relevant to `/* /index` so let's generalise it to make it relevant to all.

Something like:
```suggestion
					'Infinite loop detected in this rule and has been ignored. This will cause a redirect to strip `.html` or `/index` and end up triggering this rule again. Please fix or remove this rule to silence this warning.',
```